contrib/CollectHelper.lua

```diff
@@ -0,0 +1,199 @@
+--[[Collect Helper plugin for darktable
+
+  copyright (c) 2019  Kevin Ertel
+  
+  darktable is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  darktable is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+--[[About this plugin
+This plugin adds the button(s) to the "Selected Images" module:
+1) Return to Previous Collection
+2) Collect on image's Folder
+3) Collect on image's Color Label(s)
+4) Collect on All (AND)
+
+It also adds 3 preferences to the lua options dialog box which allow the user to activate/deactivate the 3 "Collect on" buttons.
+
+Button Behavior:
+1) Return to Previous Collection - Will reset the collect parameters to the previously active settings
+2) Collect on image's Folder - Will change the collect parameters to be "Folder" with a value of the selected image's folder location
+3) Collect on image's Color Label(s) - Will change the collect parameter to be "Color" with a value of the selected images color labels, will apply multiple parameters with AND logic if multiple exist
+4) Collect on All (AND) - Will collect on all parameters activated by the preferences dialog, as such this button is redundant if you only have one of the two other options enabled
+
+----REQUIRED SOFTWARE----
+NA
+
+----USAGE----
+Install: (see here for more detail: https://github.com/darktable-org/lua-scripts )
+ 1) Copy this file in to your "lua/contrib" folder where all other scripts reside. 
+ 2) Require this file in your luarc file, as with any other dt plug-in
+
+Select the photo you wish to change you collection based on.
+In the "Selected Images" module click on "Collect on this Image"
+
+----KNOWN ISSUES----
+]]
+
+local dt = require "darktable"
+previous = nil
+all_active = false
+
+-- FUNCTION --
+local function CheckSingleImage(selection)
+	if #selection ~= 1 then
+		dt.print("Please select a single image")
+		return 1
+	end
+	return 0
+end
+local function PreviousCollection()
+	if previous ~= nil then
+		previous = dt.gui.libs.collect.filter(previous)
+	end
+end
+local function CollectOnFolder()
+	local images = dt.gui.selection()

```

Comment:
The API manual recommends using dt.gui.action_images() instead of dt.gui.selection().

Reply:
Yea, lets talk about that real quick. Here are my thoughts:
1) I use to get some glitches when scrolling real fast through big collections, or moving my mouse in/out of the lightable grid very quickly. In these cases an image was staying hovered when it shouldn't. This was frustrating. This doesn't seem to happen anymore though.
2) Theoretically the mouse will be on the button when activated, so nothing could ever be hovered

But it boils down to: personally I get frustrated a lot when using dt by the selected/hovered behavior and feel that selected works better for this script in case of any hovered glitches. Also this script requires a single image be selected/active to run. So if I included hovered behavior than it would force the user to either hover over the selected image or none at all, so it's really only making the script potentially more restrictive on user behavior than needed. 

If you all feel very strongly this is really bad then we can change it for inclusion in the repo. (I'll probably revert it back on my local machine though)

Reply:
I've not personally seen the hover issue where an image remains hovered, but I could see it happening if a machine got busy.  Your reasoning is sound, especially about not wanting the hover, so just leave it as it is.